lib/mixins/properties-changed.html

```diff
@@ -0,0 +1,486 @@
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../utils/boot.html">
+<link rel="import" href="../utils/mixin.html">
+<link rel="import" href="../utils/async.html">
+
+<script>
+  (function () {
+
+    'use strict';
+
+    /** @const {!AsyncInterface} */
+    const microtask = Polymer.Async.microTask;
+
+    /**
+     * Element class mixin that provides basic meta-programming for creating one
+     * or more property accessors (getter/setter pair) that enqueue an async
+     * (batched) `_propertiesChanged` callback.
+     *
+     * For basic usage of this mixin, simply declare attributes to observe via
+     * the standard `static get observedAttributes()`, implement `_propertiesChanged`
+     * on the class, and then call `MyClass.createPropertiesForAttributes()` once
+     * on the class to generate property accessors for each observed attribute
+     * prior to instancing. Last, call `this._enableProperties()` in the element's
+     * `connectedCallback` to enable the accessors.
+     *
+     * Any `observedAttributes` will automatically be
+     * deserialized via `attributeChangedCallback` and set to the associated
+     * property using `dash-case`-to-`camelCase` convention.
+     *
+     * @mixinFunction
+     * @polymer
+     * @memberof Polymer
+     * @summary Element class mixin for reacting to property changes from
+     *   generated property accessors.
+     */
+    Polymer.PropertiesChanged = Polymer.dedupingMixin(superClass => {
+
+      /**
+       * @polymer
+       * @mixinClass
+       * @extends {superClass}
+       * @implements {Polymer_PropertiesChanged}
+       * @unrestricted
+       */
+      class PropertiesChanged extends superClass {
+
+        /**
+         * Creates property accessors for the given property names.
+         * @param {Object} props Object whose keys are names of accessors.
+         * @protected
+         */
+        static createProperties(props) {
+          const proto = this.prototype;
+          for (let prop in props) {
+            // don't stomp an existing accessor
+            if (!(prop in proto)) {
+              proto._createPropertyAccessor(prop);
+            }
+          }
+        }
+
+        /**
+         * Returns an attribute name that corresponds to the given property.
+         * The attribute name is the lowercased property name. Override to
+         * customize this mapping.
+         * @param {string} property Property to convert
+         * @return {string} Attribute name corresponding to the given property.
+         *
+         * @protected
+         */
+        static attributeNameForProperty(property) {
+          return property.toLowerCase();
```

Comment:
This is different to the previous mapping where it was camelcase to kebabcase right?